exercises/concept/fruit-picker/fruit-picker.spec.js

```diff
@@ -1,116 +1,51 @@
 // @ts-check
 
-import {
-  purchaseInventoryIfAvailable,
-  isServiceOnline,
-  pickAndPurchaseFruit,
-  pickFruit,
-} from './fruit-picker';
-import {
-  setStatus,
-  resetStatus,
-  setResponse,
-  getLastQuery,
-  resetQuery,
-} from './grocer';
+import { notify } from './notifier';
+import { order } from './grocer';
+import { onError, onSuccess, orderFromGrocer, postOrder } from './fruit-picker';
 
-describe('service status', () => {
-  beforeEach(() => {
-    resetStatus();
-  });
-
-  test('returns the initial status of the service', () => {
-    expect(isServiceOnline()).toBe(false);
-  });
-
-  test('returns the status when service is online', () => {
-    setStatus('ONLINE');
-    expect(isServiceOnline()).toBe(true);
-  });
-});
-
-describe('inventory service', () => {
-  beforeEach(() => {
-    resetQuery();
-  });
-
-  test('uses the query format', () => {
-    pickFruit('strawberry', 5, () => 'PURCHASE');
-    expect(getLastQuery()).toEqual({
-      variety: 'strawberry',
-      quantity: 5,
-    });
-  });
-
-  test('takes parameters for the query', () => {
-    pickFruit('blueberry', 20, () => 'PURCHASE');
-    expect(getLastQuery()).toEqual({
-      variety: 'blueberry',
-      quantity: 20,
-    });
-  });
+jest.mock('./notifier', () => ({
+  notify: jest.fn(),
+}));
 
-  test('returns synchronously', () => {
-    expect(pickFruit('melon', 1, () => 'PURCHASE')).toBe('PURCHASE');
-  });
-
-  test('returns the inventory status', () => {
-    expect(pickFruit('melon', 1, () => 'NOOP')).toBe('NOOP');
-  });
-});
-
-describe('inventory result callback', () => {
-  test('throws error if receives inventory error', () => {
-    expect(() => {
-      purchaseInventoryIfAvailable('inventory error', undefined);
-    }).toThrow();
-  });
+jest.mock('./grocer', () => ({
+  order: jest.fn(),
+}));
 
-  test('returns "PURCHASE" when inventory is available', () => {
-    expect(purchaseInventoryIfAvailable(null, true)).toBe('PURCHASE');
+describe('fruit-picker', () => {
```

Comment:
The other concept exercises use no describe block that wraps everything but instead one describe block per task. I think the association of task number to test depends on this. You can use `afterEach` outside of a describe block as well so it is not needed for that afaik. Ideally, the describe block would have the name of the function (e.g. "onSuccess") and then the `test` block is named after what is tested inside (e.g. "should call notify ...").

Reply:
That is correct. The task number can _automatically_ be inferred, _if_ it's not nested.

Reply:
So I am seeing two choices here, which would you like?

1. a describe block per task?
2. or just removing the describe block?

Reply:
The first option would be best imo.

Reply:
Yes. Describe per task is most in line with the other exercises!

Reply:
Done in https://github.com/exercism/javascript/pull/1725/commits/ea513fc44ca46d81fde40ae3c0661213dae4dea6